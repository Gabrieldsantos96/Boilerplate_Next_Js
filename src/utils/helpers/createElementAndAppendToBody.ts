export function createElementAndAppendToBody(containerId: string) {
  const wrapperElement = document.createElement('div')
  wrapperElement.setAttribute('id', containerId)
  wrapperElement.setAttribute('data-testid', containerId)
  document.body.appendChild(wrapperElement)
  return wrapperElement
}
