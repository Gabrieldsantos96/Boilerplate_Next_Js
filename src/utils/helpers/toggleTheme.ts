import EventManager from '@/lib/eventManager'

export const themeEventManager = new EventManager<string>()

export const toggleTheme = () => {
  themeEventManager.emit('toggleTheme', '')
}
