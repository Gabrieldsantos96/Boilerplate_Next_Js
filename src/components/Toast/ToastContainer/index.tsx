import { useToastContainer } from './useToastContainer'
import { toastEventManager } from '@/utils/helpers/showToast'
import ToastList from '../ToastList'

const ToastContainer = () => {
  const { handleRemoveMessage, messages } = useToastContainer({
    toastEventManager
  })
  return (
    <ToastList messages={messages} handleRemoveMessage={handleRemoveMessage} />
  )
}

export default ToastContainer
