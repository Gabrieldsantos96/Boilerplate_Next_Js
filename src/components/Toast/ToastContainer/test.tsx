import { render } from '@/utils/helpers/testUtils'

import ToastContainer from '.'
import theme from '@/styles/lightTheme'

describe('<ToastContainer />', () => {
  it('should render toast container', () => {
    const { container } = render(<ToastContainer />)

    expect(container.firstChild).toHaveStyle({
      position: 'fixed',
      left: '50%',
      'z-index': 2,
      bottom: theme.spacings.xlarge,
      transform: 'translateX(-50%)'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
