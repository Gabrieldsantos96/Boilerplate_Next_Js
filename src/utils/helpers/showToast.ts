import { EventManagerDataTransfer } from '@/components/Toast/ToastContainer/useToastContainer'
import EventManager from '@/lib/eventManager'

export const toastEventManager = new EventManager<EventManagerDataTransfer>()

export const showToast = ({
  type,
  text,
  duration
}: EventManagerDataTransfer) => {
  toastEventManager.emit('addToast', { type, text, duration })
}
