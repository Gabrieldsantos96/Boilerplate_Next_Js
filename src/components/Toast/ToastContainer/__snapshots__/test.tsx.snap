// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastContainer /> should render toast container 1`] = `
.c0 {
  position: fixed;
  bottom: 4.8rem;
  left: 50%;
  z-index: 2;
  transform: translateX(-50%);
}

<div
  class="c0"
/>
`;
